core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreUtil.java

```diff
@@ -252,4 +264,75 @@ public static void validateKeyStoreConfigName(String propertyName) throws Carbon
             throw new CarbonException("Requested key store configuration is invalid.");
         }
     }
+
+    /**
+     * Dumping the generated public cert to a file.
+     *
+     * @param configurationContext Configuration context of the current message.
+     * @param cert                 Content of the certificate.
+     * @param fileName             File name.
+     * @return file system location of the public cert.
+     */
+    public static String dumpCert(ConfigurationContext configurationContext, byte[] cert, String fileName) {
+
+        if (!verifyCertExistence(fileName, configurationContext)) {
+            String workingDirectory = (String) configurationContext.getProperty(ServerConstants.WORK_DIR);
+            File publicCert = new File(workingDirectory + File.separator + ServerConstants.PUBLIC_CERTS_DIRECTORY_NAME);
+
+            if (fileName == null) {
+                fileName = System.currentTimeMillis() + new SecureRandom().nextDouble() + ".cert";
+            }
+            if (!publicCert.exists()) {
+                publicCert.mkdirs();
+            }
+
+            String filePath = workingDirectory + File.separator + ServerConstants.PUBLIC_CERTS_DIRECTORY_NAME +
+                    File.separator + fileName;
+            try (OutputStream outStream = Files.newOutputStream(Paths.get(filePath))) {
+                outStream.write(cert);
+            } catch (Exception e) {
+                String msg = "Error when writing the public certificate to a file";
+                log.error(msg, e);
+                throw new SecurityException(msg, e);
+            }
+
+            Map fileResourcesMap = (Map) configurationContext.getProperty(ServerConstants.FILE_RESOURCE_MAP);
+            if (fileResourcesMap == null) {
+                fileResourcesMap = new Hashtable();
+                configurationContext.setProperty(ServerConstants.FILE_RESOURCE_MAP, fileResourcesMap);
+            }
+            fileResourcesMap.put(fileName, filePath);
+        }
+        return ServerConstants.ContextPaths.DOWNLOAD_PATH + "?id=" + fileName;
+    }
+
+    /**
+     * Check whether the certificate is available in the file system.
+     *
+     * @param fileName             File name.
+     * @param configurationContext Configuration context of the current message.
+     * @return True if the certificate is available in the file system, false otherwise.
+     */
+    private static boolean verifyCertExistence(String fileName, ConfigurationContext configurationContext) {
+
+        String workingDirectory = (String) configurationContext.getProperty(ServerConstants.WORK_DIR);
+        String filePath =
+                workingDirectory + File.separator + ServerConstants.PUBLIC_CERTS_DIRECTORY_NAME + File.separator +
+                        fileName;
+        File pubCert = new File(
```

Comment:
```suggestion
        File pubCert = new File(filePath);
```

Reply:
Removed this method in the latest cahnges